File: XUCore/additions/NSURL+NSURLAdditions.m
// 
// NSURL+NSURLAdditions.m
// 
// Created by Charlie Monroe
// 
// Copyright (c) 2010-13 Fuel Collective, LLC. All rights reserved.
// 


#import "NSURL+NSURLAdditions.h"
#import "NSStringAdditions.h"

@implementation NSURL (NSURLAdditions)

+(id)fileReferenceURLWithPath:(NSString *)path{
	if (path == nil){
		return nil;
	}
	NSURL *fileURL = [self fileURLWithPath:path];
	NSURL *referenceURL = [fileURL fileReferenceURL];
	return (referenceURL == nil ? fileURL : referenceURL);
}

-(NSInteger)fileSize{
	if (![self isFileURL]){
		return 0;
	}
	return [[[[NSFileManager defaultManager] attributesOfItemAtPath:[self path] error:NULL] objectForKey:NSFileSize] integerValue];
}
-(NSString *)fileSizeString{
	if (![self isFileURL]){
		return [NSString stringWithFormat:@"Not Applicable (%@)", self];
	}
	
	return [NSByteCountFormatter stringFromByteCount:[self fileSize] countStyle:NSByteCountFormatterCountStyleFile];
}

@end

File: XUCore/additions/NSURL+NSURLAdditions.swift
//
//  NSURL+NSURLAdditions.swift
//  XUCore
//
//  Created by Charlie Monroe on 11/10/15.
//  Copyright © 2015 Charlie Monroe Software. All rights reserved.
//

import Foundation

public extension URL {
	
	@available(*, deprecated)
	public static func fileReferenceURLWithPath(_ path: String) -> URL? {
		let fileURL = URL(fileURLWithPath: path)
		return (fileURL as NSURL).fileReferenceURL()
	}
	
	@available(*, deprecated)
	public var fileSizeString: String {
		if !self.isFileURL {
			return "Not Applicable (\(self))"
		}
		
		return ByteCountFormatter.string(fromByteCount: Int64(self.fileSize), countStyle: .file)
	}
	
}


